pkg/kubelet/network/dns/dns.go

```diff
@@ -99,19 +102,31 @@ func omitDuplicates(strs []string) []string {
 func (c *Configurer) formDNSSearchFitsLimits(composedSearch []string, pod *v1.Pod) []string {
 	limitsExceeded := false
 
-	if len(composedSearch) > validation.MaxDNSSearchPaths {
-		composedSearch = composedSearch[:validation.MaxDNSSearchPaths]
```

Comment:
I think you should still keep something like this - before you iterate aan unbounded list - but truncate to the new, larger value and flag limitsExceeded here